File: RayTracerVBSingleProj/Scene.vb
Imports System.Linq

Public Class Scene

    Public Property Things As Thing()

    Public Property Lights As LightSource()

    Public Property Camera As Camera

    Public Sub New(ByVal things As Thing(), ByVal lights As LightSource(), ByVal camera As Camera)
        Me.Things = things
        Me.Lights = lights
        Me.Camera = camera
    End Sub

    Public Function Intersect(ByVal r As Ray) As IEnumerable(Of Intersection)
        Return Things.Select(Function(t) t.CalculateIntersection(r))
    End Function
End Class

File: RayTracerCSSingleProj/Scene.cs
using System.Collections.Generic;
using System.Linq;

namespace RayTracer
{
    public class Scene
    {
        public Thing[] Things { get; private set; }
        public LightSource[] Lights { get; private set; }
        public Camera Camera { get; private set; }

        public Scene(Thing[] things, LightSource[] lights, Camera camera)
        {
            Things = things;
            Lights = lights;
            Camera = camera;
        }

        public List<Intersection> Intersect(Ray r)
        {
            return Things.Select(t => t.CalculateIntersection(r)).ToList();
        }
    }
}